### PR TITLE
[Cherry-pick][Move-package] Enable bytecode as dependencies

### PR DESCRIPTION
**Pull request by rahxephon89**
## Motivation

Cherry-pick #865 onto `main`

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes

### CODE REVIEW

**Review by wrwg — APPROVED** [review]
Should have someone from Mysten to approve as well.

File: language/tools/move-package/src/compilation/compiled_package.rs [hunk 1]
```diff
@@ -575,11 +578,27 @@ impl CompiledPackage {
...
+        let (src_deps, bytecode_deps): (Vec<_>, Vec<_>) = deps_package_paths
+            .clone()
+            .into_iter()
+            .partition_map(|(p, b)| if b { Either::Left(p) } else { Either::Right(p) });
+        // If bytecode dependency is not empty, do not allow renaming
+        if !bytecode_deps.is_empty() {
+            if let Some(pkg_name) = resolution_graph.contains_renaming() {
+                anyhow::bail!(
+                    "Found address renaming in package '{}' when \
+                    building Move model -- this is currently not supported",
```
Comment (amnn) [inline language/tools/move-package/src/compilation/compiled_package.rs:591]: nit: This code is not related to Move model, so we should update this error message.

```suggestion
                    building with bytecode dependencies -- this is currently not supported",
```

**Review by amnn — APPROVED** [review]
👋 Mysten person here.

Thanks for contributing this!  One small nit from our side, on one of the error messages, but this change otherwise looks good.  It's going to conflict slightly with the lock files changes, but not in a fundamental way.
